### DISCUSSION TITLE
CS Tools installation using offline installer

### DISCUSSION THREAD

**Discussion by sae-bruce** [thoughtspot/cs_tools#77] [General] [answered]
C:\Users\m172485>python -m sysconfig
Platform: "win-amd64"
Python version: "3.7"
Current installation scheme: "nt"

Paths:
        data = "C:\Users\m172485\Anaconda3"
        include = "C:\Users\m172485\Anaconda3\Include"
        platinclude = "C:\Users\m172485\Anaconda3\Include"
        platlib = "C:\Users\m172485\Anaconda3\Lib\site-packages"
        platstdlib = "C:\Users\m172485\Anaconda3\Lib"
        purelib = "C:\Users\m172485\Anaconda3\Lib\site-packages"
        scripts = "C:\Users\m172485\Anaconda3\Scripts"
        stdlib = "C:\Users\m172485\Anaconda3\Lib"

Variables:
        BINDIR = "C:\Users\m172485\Anaconda3"
        BINLIBDEST = "C:\Users\m172485\Anaconda3\Lib"
        EXE = ".exe"
        EXT_SUFFIX = ".pyd"
        INCLUDEPY = "C:\Users\m172485\Anaconda3\Include"
        LIBDEST = "C:\Users\m172485\Anaconda3\Lib"
        SO = ".pyd"
        VERSION = "37"
        abiflags = ""
        base = "C:\Users\m172485\Anaconda3"
        exec_prefix = "C:\Users\m172485\Anaconda3"
        installed_base = "C:\Users\m172485\Anaconda3"
        installed_platbase = "C:\Users\m172485\Anaconda3"
        platbase = "C:\Users\m172485\Anaconda3"
        prefix = "C:\Users\m172485\Anaconda3"
        projectbase = "C:\Users\m172485\Anaconda3"
        py_version = "3.7.7"
        py_version_nodot = "37"
        py_version_short = "3.7"
        srcdir = "C:\Users\m172485\Anaconda3"
        userbase = "C:\Users\m172485\AppData\Roaming\Python"

**Comment by boonhapus** [comment, marked as answer]
Hi @sae-bruce , I'm having a hard time building an offline installer for py3.7 , are you able to upgrade to a more recent version? Python 3.10 or 3.11 is preferred. 3.12 is still too new for us as well.

**Reply by sae-bruce** [reply]
Hi @boonhapus , I created a virtual environment with python 3.10 by conda, following is the output of the command "python -m sysconfig", please check if it's suitable.

(cs_tool) C:\Users\m172485>python -m sysconfig
Platform: "win-amd64"
Python version: "3.10"
Current installation scheme: "nt"

Paths:
        data = "C:\Users\m172485\Anaconda3\envs\cs_tool"
        include = "C:\Users\m172485\Anaconda3\envs\cs_tool\Include"
        platinclude = "C:\Users\m172485\Anaconda3\envs\cs_tool\Include"
        platlib = "C:\Users\m172485\Anaconda3\envs\cs_tool\Lib\site-packages"
        platstdlib = "C:\Users\m172485\Anaconda3\envs\cs_tool\Lib"
        purelib = "C:\Users\m172485\Anaconda3\envs\cs_tool\Lib\site-packages"
        scripts = "C:\Users\m172485\Anaconda3\envs\cs_tool\Scripts"
        stdlib = "C:\Users\m172485\Anaconda3\envs\cs_tool\Lib"

Variables:
        BINDIR = "C:\Users\m172485\Anaconda3\envs\cs_tool"
        BINLIBDEST = "C:\Users\m172485\Anaconda3\envs\cs_tool\Lib"
        EXE = ".exe"
        EXT_SUFFIX = ".cp310-win_amd64.pyd"
        INCLUDEPY = "C:\Users\m172485\Anaconda3\envs\cs_tool\Include"
        LIBDEST = "C:\Users\m172485\Anaconda3\envs\cs_tool\Lib"
        SO = ".cp310-win_amd64.pyd"
        TZPATH = "C:\Users\m172485\Anaconda3\envs\cs_tool\share\zoneinfo"
        VERSION = "310"
        abiflags = ""
        base = "C:\Users\m172485\Anaconda3\envs\cs_tool"
        exec_prefix = "C:\Users\m172485\Anaconda3\envs\cs_tool"
        installed_base = "C:\Users\m172485\Anaconda3\envs\cs_tool"
        installed_platbase = "C:\Users\m172485\Anaconda3\envs\cs_tool"
        platbase = "C:\Users\m172485\Anaconda3\envs\cs_tool"
        platlibdir = "lib"
        prefix = "C:\Users\m172485\Anaconda3\envs\cs_tool"
        projectbase = "C:\Users\m172485\Anaconda3\envs\cs_tool"
        py_version = "3.10.13"
        py_version_nodot = "310"
        py_version_nodot_plat = "310"
        py_version_short = "3.10"
        srcdir = "C:\Users\m172485\Anaconda3\envs\cs_tool"
        userbase = "C:\Users\m172485\AppData\Roaming\Python"

(cs_tool) C:\Users\m172485>

**Reply by boonhapus** [reply]
Thanks @sae-bruce ! I've actually never had someone try this in a conda env, will be keen to see how this works out for you. It should install fine, but potentially the global `cs_tools` command might not work for you. CS Tools creates and manages its own virtual environment, which is what `conda` is already providing you.

Let me know how it goes!

---

1. Extract the zip file.
2. Run python against the `_bootstrapper.py` file with `--offline-mode` specified

`cd cs-tools_1.4.13_win-amd64_3.10`
`python _bootstrapper.py --install --offline-mode .`

It's a bit too big for GitHub discussions, so I've attached an Egnyte link. Let me know if that works for you!

**[cs-tools_1.4.13_win-amd64_3.10.zip](https://thoughtspot.egnyte.com/dl/rCHRjaRTez/cs-tools_1.4.13_win-amd64_3.10.zip_)**

**Reply by sae-bruce** [reply]
Hi @boonhapus , I executed the command, but it returned error.

(cs_tool) PS D:\M172485\WorkProject\Cloud Migration\cs-tools_1.4.13_win-amd64_3.10> python _bootstrapper.py --install --offline-mode .
13:07:20 | Welcome to the CS Tools Bootstrapper!
         | Ideally, you will only need to run this one time, and then your environment can be fully managed by CS Tools itself.
         | If you run into any issues, please reach out to us on GitHub Discussions below.
         |
         |           GitHub: https://github.com/thoughtspot/cs_tools/issues/new/choose
         |
13:07:21 | Using the offline binary found at .
         | Installing CS Tools and its dependencies.
13:07:22 | Looking in links: .
13:07:23 | Collecting cs_tools[cli]
13:07:24 | Installing build dependencies: started
13:07:27 | Installing build dependencies: finished with status 'error'
         | Command errored out with exit status 1:
         | command: 'C:\Users\m172485\AppData\Roaming\cs_tools\.cs_tools\Scripts\python.exe' 'C:\Users\m172485\AppData\Roaming\cs_tools\.cs_tools\lib\site-packages\pip' install --ignore-installed --no-user --prefix 'C:\Users\m172485\AppData\Local\Temp\pip-build-env-1c0q_47o\overlay' --no-warn-script-location --no-binary :none: --only-binary :none: --no-index --find-links . --trusted-host files.pythonhost.org --trusted-host pypi.org --trusted-host pypi.python.org --trusted-host github.com --trusted-host codeload.github.com -- 'setuptools>=42'
         | cwd: None
         | Complete output (3 lines):
         | Looking in links: .
         | Collecting setuptools>=42
         | Package 'setuptools' requires a different Python: 3.7.7 not in '>=3.8'
         | ----------------------------------------
         | Command errored out with exit status 1: 'C:\Users\m172485\AppData\Roaming\cs_tools\.cs_tools\Scripts\python.exe' 'C:\Users\m172485\AppData\Roaming\cs_tools\.cs_tools\lib\site-packages\pip' install --ignore-installed --no-user --prefix 'C:\Users\m172485\AppData\Local\Temp\pip-build-env-1c0q_47o\overlay' --no-warn-script-location --no-binary :none: --only-binary :none: --no-index --find-links . --trusted-host files.pythonhost.org --trusted-host pypi.org --trusted-host pypi.python.org --trusted-host github.com --trusted-host codeload.github.com -- 'setuptools>=42' Check the logs for full command output.
         | Unexpected error in bootstrapper, see D:\M172485\WorkProject\Cloud Migration\cs-tools_1.4.13_win-amd64_3.10\cs_tools-bootstrap-error-tmp1ovm6xyn.log for details..
(cs_tool) PS D:\M172485\WorkProject\Cloud Migration\cs-tools_1.4.13_win-amd64_3.10>

**Reply by boonhapus** [reply]
ACK @sae-bruce - let's get a call together to work through this one. I think the conda env complicates things a bit, so we might need to tinker with it. I'll reach out via email.
